From 49ddd814ac1071af7c4f45d5f8398b91862f2c7f Mon Sep 17 00:00:00 2001
From: Vladimir Mezentsev <vladimir.mezentsev@oracle.com>
Date: Wed, 17 Aug 2022 19:55:23 -0700
Subject: [PATCH] gprofng: fix bug 29479 Collection fails when built without
 java support

gprofng/ChangeLog
2022-08-17  Vladimir Mezentsev  <vladimir.mezentsev@oracle.com>

	PR gprofng/29479
	* libcollector/collector.c: Add #if defined(GPROFNG_JAVA_PROFILING) for
	java specific code.
	* libcollector/unwind.c: Likewise.
---
 gprofng/libcollector/collector.c | 2 ++
 gprofng/libcollector/unwind.c    | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/gprofng/libcollector/collector.c b/gprofng/libcollector/collector.c
index 93c9d3330de5..ceff2c1caccb 100644
--- a/gprofng/libcollector/collector.c
+++ b/gprofng/libcollector/collector.c
@@ -913,8 +913,10 @@ __collector_open_experiment (const char *exp, const char *params, sp_origin_t or
   __collector_ext_unwind_key_init (1, NULL);
 
   /* start java attach if suitable */
+#if defined(GPROFNG_JAVA_PROFILING)
   if (exp_origin == SP_ORIGIN_DBX_ATTACH)
     __collector_jprofile_start_attach ();
+#endif
   start_sec_time = CALL_UTIL (time)(NULL);
   __collector_start_time = collector_interface.getHiResTime ();
   TprintfT (DBG_LT0, "\t__collector_open_experiment; resetting start_time\n");
diff --git a/gprofng/libcollector/unwind.c b/gprofng/libcollector/unwind.c
index 119243788020..f8e11823c81e 100644
--- a/gprofng/libcollector/unwind.c
+++ b/gprofng/libcollector/unwind.c
@@ -557,6 +557,7 @@ __collector_get_frame_info (hrtime_t ts, int mode, void *arg)
   int size = max_frame_size;
 
 #define MIN(a,b) ((a)<(b)?(a):(b))
+#if defined(GPROFNG_JAVA_PROFILING)
   /* get Java info */
   if (__collector_java_mode && __collector_java_asyncgetcalltrace_loaded && context && !pseudo_context)
     {
@@ -569,6 +570,7 @@ __collector_get_frame_info (hrtime_t ts, int mode, void *arg)
 	  size -= sz;
 	}
     }
+#endif
 
   /* get native stack */
   if (context)
